xbmc/addons/addoninfo/AddonInfoBuilder.cpp

```diff
@@ -354,8 +355,26 @@ bool CAddonInfoBuilder::ParseXML(const AddonInfoPtr& addon, const TiXmlElement*
       if (element && element->GetText() != nullptr)
         addon->m_packageSize = StringUtils::ToUint64(element->GetText(), 0);
 
-      /* Parse addon.xml "<news lang="..">...</news>" */
+      /* Parse addon.xml "<news lang="..">...</news>"
+       *
+       * In the event that the changes log (news) in addon.xml is empty, check
+       * whether it is an installed addon and read a changelog.txt as a
+       * replacement, if available. */
       GetTextList(child, "news", addon->m_changelog);
+      if (addon->m_changelog.empty() && repo.artdir.empty() && !addonPath.empty())
```

Comment:
This field is only used for repos, so if something is there, changelog.txt can never be present.
If this were not possible, it would try to find the "changelog.txt" in the network for every addon

Reply:
Added a new commit, which will set it to a new value. 
I don't want to do it as an extra function variable because `const CRepository::DirInfo& repo` is only set if it comes from the repo anyway.